metric/metricsController.go

```diff
@@ -0,0 +1,96 @@
+package metric
+
+import (
+	prom "github.com/prometheus/client_golang/prometheus"
+)
+
+type MetricsCtl struct {
+	subsystem string
+	counters  map[string]prom.Counter
+	gauges    map[string]prom.Gauge
+}
+
+func New(subsystem string) *MetricsCtl {
+	ctl := &MetricsCtl{
+		subsystem: subsystem,
+		counters:  make(map[string]prom.Counter),
+		gauges:    make(map[string]prom.Gauge),
+	}
+	ctl.RegisterCounter(unregisteredCounter, "Counter for unregistered metrics")
+	ctl.RegisterGauge(unregisteredGauge, "Gauge for unregistered metrics")
+	return ctl
+}
+
+func (mc *MetricsCtl) RegisterCounter(name, help string) {
+	RegisterCounter(&MetricDesc{
+		Subsystem: mc.subsystem,
+		Name:      name,
+		Help:      help,
+	})
+	mc.counters[name] = GetCounter(mc.subsystem, name)
```

Comment:
Same here: GetCounter, RegisterCounter, RegisterGauge, RegisterGauge should be removed